Utilities/bootstrap

```diff
@@ -184,6 +184,7 @@ def parse_global_args(args):
     args.source_dirs["tsc"]                   = os.path.join(args.project_root, "..", "swift-tools-support-core")
     args.source_dirs["yams"]                  = os.path.join(args.project_root, "..", "yams")
     args.source_dirs["swift-argument-parser"] = os.path.join(args.project_root, "..", "swift-argument-parser")
+    args.source_dirs["swift-crypto"]          = os.path.join(args.project_root, "..", "swift-crypto")
```

Comment:
@neonichu I think we have to add `swift-crypto` back since `PackageSigning` depends on it and `PackageRegistry` depends on `PackageSigning`.

Reply:
Yep, I think that's true for now. Eventually, `PackageRegistry` shouldn't build with CMake at all since it isn't needed for bootstrapping.

Reply:
Windows can't bootstrap until we fix the myriad of issues around linking in SPM.  We should be adding it to CMake.

Reply:
> We should be adding it to CMake.

I thought I did with https://github.com/apple/swift-package-manager/pull/6184/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a, or there is something else? The Windows CI is failing with this:

```
FAILED: lib/SKSwiftPMWorkspace.lib Sources/SKSwiftPMWorkspace/CMakeFiles/SKSwiftPMWorkspace.dir/SwiftPMWorkspace.swift.obj swift/SKSwiftPMWorkspace.swiftmodule 
cmd.exe /C "cd . && T:\1\bin\swiftc.exe -output-file-map Sources\SKSwiftPMWorkspace\CMakeFiles\SKSwiftPMWorkspace.dir\Release\output-file-map.json -incremental -j 36 -emit-library -static -o lib\SKSwiftPMWorkspace.lib -module-name SKSwiftPMWorkspace -module-link-name SKSwiftPMWorkspace -emit-module -emit-module-path swift\SKSwiftPMWorkspace.swiftmodule -emit-dependencies  -O -libc MD -Xcc -D_CRT_SECURE_NO_WARNINGS -Xcc -FT:/4 -autolink-force-load -Xfrontend -validate-tbd-against-ir=none -I swift -I T:\7\swift -I T:\6\swift -I C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift-system\Sources\CSystem\include -I T:\4\swift -I T:\3\src\swift\swift -I T:\3 -I C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift-corelibs-libdispatch -I C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift-corelibs-libdispatch\src -I T:\3\src -I C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift-corelibs-libdispatch\src\BlocksRuntime -I T:\14\swift -I T:\13\swift -I T:\8\products\llbuildSwift -I C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\llbuild\products\libllbuild\include C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\sourcekit-lsp\Sources\SKSwiftPMWorkspace\SwiftPMWorkspace.swift    && cd ."
<unknown>:0: error: missing required module 'Crypto'
```

Reply:
If we use a non-`@_implementationOnly` import and the module is transitively part of libSwiftPM, we'll need to install it.

Reply:
> Windows can't bootstrap until we fix the myriad of issues around linking in SPM. We should be adding it to CMake.

I don't disagree with this, but we should be careful about what's *essential* to be build with CMake and what isn't, because otherwise we may end up with ancillary stuff being required for bootstrapping. This goes beyond Windows, btw, e.g. right now we need to build `PackagePlugin` with CMake because we weren't conscious of this in the past and haven't fixed it, yet.

Reply:
> I don't disagree with this, but we should be careful about what's _essential_ to be build with CMake and what isn't, because otherwise we may end up with ancillary stuff being required for bootstrapping. This goes beyond Windows, btw, e.g. right now we need to build `PackagePlugin` with CMake because we weren't conscious of this in the past and haven't fixed it, yet.

Except we need the full build to distribute SPM on Windows.  Otherwise, we end up without support for SPM on Windows.

Reply:
> Except we need the full build to distribute SPM on Windows. Otherwise, we end up without support for SPM on Windows.

Yah, that's why we are keeping the full build for now and the foreseeable future. That doesn't mean we shouldn't be aware of the fact that the way it is supposed to work in the fullness of time is not having everything build with CMake.